src/docfx/lib/log/ErrorLog.cs

```diff
@@ -217,40 +223,50 @@ private int GetMaxCount(Config? config, ErrorLevel level)
 
             return level switch
             {
-                ErrorLevel.Error => config.MaxErrors,
-                ErrorLevel.Warning => config.MaxWarnings,
-                ErrorLevel.Suggestion => config.MaxSuggestions,
+                ErrorLevel.Error => config.MaxFileErrors,
+                ErrorLevel.Warning => config.MaxFileWarnings,
+                ErrorLevel.Suggestion => config.MaxFileSuggestions,
+                ErrorLevel.Info => config.MaxFileInfos,
                 _ => int.MaxValue,
             };
         }
 
-        private bool ExceedMaxErrors(Config? config, ErrorLevel level)
+        private bool ExceedMaxErrors(Config? config, ErrorLevel level, FilePath? filePath)
         {
-            if (config == null)
+            if (filePath == null)
             {
                 return false;
             }
 
-            return level switch
+            if (TryGetFileCount(level, out var fileCount))
             {
-                ErrorLevel.Error => Volatile.Read(ref _errorCount) >= config.MaxErrors,
-                ErrorLevel.Warning => Volatile.Read(ref _warningCount) >= config.MaxWarnings,
-                ErrorLevel.Suggestion => Volatile.Read(ref _suggestionCount) >= config.MaxSuggestions,
-                ErrorLevel.Info => Volatile.Read(ref _infoCount) >= config.MaxInfos,
-                _ => false,
-            };
+                return fileCount.GetValueOrDefault(filePath, 0) >= GetFileMaxCount(config, level);
+            }
+
+            return false;
         }
 
-        private bool IncrementExceedMaxErrors(Config? config, ErrorLevel level)
+        private bool IncrementExceedMaxErrors(Config? config, ErrorLevel level, FilePath? filePath)
         {
-            return level switch
+            if (filePath == null)
             {
-                ErrorLevel.Error => Interlocked.Increment(ref _errorCount) > (config?.MaxErrors ?? int.MaxValue),
-                ErrorLevel.Warning => Interlocked.Increment(ref _warningCount) > (config?.MaxWarnings ?? int.MaxValue),
-                ErrorLevel.Suggestion => Interlocked.Increment(ref _suggestionCount) > (config?.MaxSuggestions ?? int.MaxValue),
-                ErrorLevel.Info => Interlocked.Increment(ref _infoCount) > (config?.MaxInfos ?? int.MaxValue),
-                _ => false,
-            };
+                return false;
+            }
+
+            if (TryGetFileCount(level, out var fileCount))
+            {
+                return fileCount.AddOrUpdate(filePath, 1, (_, oldCount) => ++oldCount) > GetFileMaxCount(config, level);
```

Comment:
>fileCount [](start = 23, length = 9)

for each file, the errors are reported sequently, do we have a light way to calculate the error count per file?

Reply:
Because there may be mutiple files include one file, and the included file contains some validation error. In this case log is parallel.